### DISCUSSION TITLE
required aws-sdk-cpp version is out of date

### DISCUSSION THREAD

**Discussion by adonishong** [milvus-io/milvus#20876] [Q&A and General discussion]
Issue:
The aws-sdk-cpp version (1.9.234) required by "internal/core/conanfile.txt" is out of date, the sprintf call from "aws-cpp-sdk-core/source/external/cjson/cJSON.cpp" will break the build progress.

Solution:
1. Update aws-sdk-cpp version to latest version should fix issue, 
2. Create a patch to replace all sprintf call to snprintf call inside "aws-cpp-sdk-core/source/external/cjson/cJSON.cpp" should work too

Work around
B4 this issue been fixed, replace cJSON.cpp during build progress should work

**Comment by yhmo** [comment]
I guess your OS is MacOS Ventura?
I searched the issues of aws-sdk-cpp, looks like this issue is what you are talking: https://github.com/aws/aws-sdk-cpp/issues/2195

Currently, this version works well in our environment(mainly centos/ubuntu), we didn't encounter this problem.
